shell/src/main/java/alluxio/cli/ValidateEnv.java

```diff
@@ -0,0 +1,392 @@
+        MASTERS_SSH_VALIDATION_TASK,
+        WORKERS_SSH_VALIDATION_TASK
+    ));
+    targetMap.put("local", TASK_NAMES.keySet());
+    return targetMap;
+  }
+
+  private interface ValidationTask {
+    boolean validate();
+  }
+
+  private static ValidationTask create(String name, ValidationTask task) {
+    TASK_NAMES.put(task, name);
+    return task;
+  }
+
+  private static boolean isLocalPortAvailable(int port) {
+    try (ServerSocket socket = new ServerSocket(port)) {
+      socket.setReuseAddress(true);
+      return true;
+    } catch (IOException e) {
+      return false;
+    }
+  }
+
+  private static boolean isAddressReachable(String hostname, int port) {
+    try (Socket socket = new Socket(hostname, port)) {
+      return true;
+    } catch (IOException e) {
+      return false;
+    }
+  }
+
+  private static boolean isAlluxioRunning(String className) {
+    String[] command = {"bash", "-c",
+                        "ps -Aww -o command | grep -i \"[j]ava\" | grep " + className};
+    try {
+      Process p = Runtime.getRuntime().exec(command);
+      try (InputStreamReader input = new InputStreamReader(p.getInputStream())) {
+        if (input.read() >= 0) {
+          return true;
+        }
+      }
+      return false;
+    } catch (IOException e) {
+      System.err.format("Unable to check Alluxio status: %s.%n", e.getMessage());
+      return false;
+    }
+  }
+
+  private static class PortAvailabilityValidationTask implements ValidationTask {
+    private final ServiceType mServiceType;
+    private final Class mOwner;
+
+    public PortAvailabilityValidationTask(ServiceType serviceType, Class owner) {
+      mServiceType = serviceType;
+      mOwner = owner;
+    }
+
+    @Override
+    public boolean validate() {
+      if (isAlluxioRunning(mOwner.getCanonicalName())) {
+        System.out.format("%s is already running. Skip validation.%n", mOwner.getSimpleName());
+        return true;
+      }
+      int port = NetworkAddressUtils.getPort(mServiceType);
+      if (!isLocalPortAvailable(port)) {
+        System.err.format("%s port %d is not available.%n", mServiceType.getServiceName(), port);
+        return false;
+      }
+      return true;
+    }
+  }
+
+  private static class SshValidationTask implements ValidationTask {
+    private final String mFileName;
+
+    public SshValidationTask(String fileName) {
+      mFileName = fileName;
+    }
+
+    @Override
+    public boolean validate() {
+      List<String> nodes = readNodeList(mFileName);
+      if (nodes == null) {
+        return false;
+      }
+
+      boolean hasUnreachableNodes = false;
+      for (String nodeName : nodes) {
+        if (!isAddressReachable(nodeName, 22)) {
+          System.err.format("Unable to reach ssh port 22 on node %s.%n", nodeName);
+          hasUnreachableNodes = true;
+        }
+      }
+      return !hasUnreachableNodes;
+    }
+  }
+
+  // read a list of nodes from given file name ignoring comments and empty lines
```

Comment:
comment on the expected format the of file, for example, is the file expected to be `conf/masters` and `conf/workers`, if so, mention this in the comment, otherwise, I'm confused what the file can be.